From 317bb0561baa7b3d1d8927c21d912b26d2719b8e Mon Sep 17 00:00:00 2001
From: Vihang Karajgaonkar <vihang@cloudera.com>
Date: Fri, 22 Jul 2016 13:24:23 -0700
Subject: [PATCH 0700/1164] Revert "CDH-42508 : Revert "CDH-42361 :
 HIVE-10624: Update the initial script to make
 beeline bucked cli as default and allow user
 choose old hive cli by env (Ferdinand via
 Xuefu)""

HIVE-10624 introduced a bug which was reported in CDH-42508. The change was reverted earlier to unblock others. Now this change reintroduces it. The bug is fixed in HIVE-12246 which needs to be backported along with this change

This reverts commit c536c1ec159af1b5667d750f2da1fc912df7e950.

Change-Id: I1bf0c56233d9903b4d12f9bbe18dfdddf6b214ba
---
 bin/ext/cli.sh              |   11 +++++------
 bin/ext/util/execHiveCmd.sh |    6 ++++--
 2 files changed, 9 insertions(+), 8 deletions(-)

diff --git a/bin/ext/cli.sh b/bin/ext/cli.sh
index ce26042..ef0e8d0 100644
--- a/bin/ext/cli.sh
+++ b/bin/ext/cli.sh
@@ -28,12 +28,11 @@ updateCli() {
 }
 
 cli () {
-  CLASS=org.apache.hadoop.hive.cli.CliDriver
-  execHiveCmd $CLASS "$@"
+  updateCli
+  execHiveCmd $CLASS $JAR "$@"
 }
 
 cli_help () {
-  CLASS=org.apache.hadoop.hive.cli.CliDriver
-  execHiveCmd $CLASS "--help"
-} 
-
+  updateCli
+  execHiveCmd $CLASS $JAR "--help"
+}
\ No newline at end of file
diff --git a/bin/ext/util/execHiveCmd.sh b/bin/ext/util/execHiveCmd.sh
index 167cc40..e46ec3c 100644
--- a/bin/ext/util/execHiveCmd.sh
+++ b/bin/ext/util/execHiveCmd.sh
@@ -16,9 +16,11 @@
 execHiveCmd () {
   CLASS=$1;
   shift;
+  JAR=$1
+  shift;
 
   # cli specific code
-  if [ ! -f ${HIVE_LIB}/hive-cli-*.jar ]; then
+  if [ ! -f ${HIVE_LIB}/$JAR ]; then
     echo "Missing Hive CLI Jar"
     exit 3;
   fi
@@ -28,5 +30,5 @@ execHiveCmd () {
   fi
 
   # hadoop 20 or newer - skip the aux_jars option. picked up from hiveconf
-  exec $HADOOP jar ${HIVE_LIB}/hive-cli-*.jar $CLASS $HIVE_OPTS "$@"
+  exec $HADOOP jar ${HIVE_LIB}/$JAR $CLASS $HIVE_OPTS "$@"
 }
-- 
1.7.9.5

